main.js

```diff
@@ -176,22 +176,38 @@ define([
                         return;
                     }
 
-                    if (mapLayer instanceof TileLayer || mapLayer instanceof MapImageLayer) {
+                    if (mapLayer instanceof TileLayer) {
                         if (layerServiceIds.length === 0) {
                             mapLayer.visible = false;
                         } else {
                             mapLayer.visible = true;
                         }
                     } else {
-                        // `.setVisibleLayers` seems to be unavailable in
```

Comment:
Should the `else` clause here apply only to `MapImageLayer` types? Or will this toggling work for other service types?